File: LoopsBitManipulationNumberBasedProblems/prime1.java
import java.util.Scanner;

public class prime1 {
    public static void main(String[] args) {
       Scanner sc=new Scanner(System.in);

       int n=sc.nextInt();
       int flag=1;

       if(n<=1)
       {
        System.out.println("not prime");
        sc.close();
        return;
       }
       for(int i=2;i<=Math.sqrt(n);i++){
        if(n%i==0){
        flag=0;
        break;
       }
    //    for(int i=2;i*i<=n;i++){
    //     if(n%i==0){
    //     flag=0;
    //    }
    }
    if(flag==1){
        System.out.println("prime number");
    }
    else
    System.out.println("not a prime");
    
    sc.close();
    }
  
}

File: BasicsofCOptimization/prime1.c
#include <stdio.h>
#include <math.h>

int main() {
    int n, flag = 1;
    scanf("%d", &n);
    
    if (n <= 1) {
        flag = 0;
    } else {
        for (int i = 2; i <= sqrt(n); i++) {
            if (n % i == 0) {
                flag = 0;
                break;
            }
        }
    }

    if (flag) {
        printf("Prime\n");
    } else {
        printf("Not Prime\n");
    }

    return 0;
}
